sdk_contrib/async_processor/src/main/java/io/opentelemetry/sdk/contrib/trace/export/DisruptorEventQueue.java

```diff
@@ -0,0 +1,217 @@
+import io.opentelemetry.sdk.trace.SpanProcessor;
+import java.util.List;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * A low-latency event queue for background updating of (possibly contended) objects. This is
+ * intended for use by instrumentation methods to ensure that they do not block foreground
+ * activities.
+ */
+@ThreadSafe
+final class DisruptorEventQueue {
+  private static final Logger logger = Logger.getLogger(DisruptorEventQueue.class.getName());
+  private static final String WORKER_THREAD_NAME = "DisruptorEventQueue_WorkerThread";
+
+  // The event queue is built on this {@link Disruptor}.
+  private final Disruptor<DisruptorEvent> disruptor;
+  private final RingBuffer<DisruptorEvent> ringBuffer;
+  private final List<SpanProcessor> spanProcessorList;
+  private final AtomicBoolean loggedShutdownMessage = new AtomicBoolean(false);
+  private volatile boolean isShutdown = false;
+  private final boolean blocking;
+
+  // Creates a new EventQueue. Private to prevent creation of non-singleton instance.
+  private DisruptorEventQueue(
+      Disruptor<DisruptorEvent> disruptor,
+      List<SpanProcessor> spanProcessorList,
+      boolean blocking) {
+    this.disruptor = disruptor;
+    this.ringBuffer = disruptor.getRingBuffer();
+    this.spanProcessorList = spanProcessorList;
+    this.blocking = blocking;
+  }
+
+  // Creates a new EventQueue. Private to prevent creation of non-singleton instance.
+  static DisruptorEventQueue create(
+      int bufferSize,
+      WaitStrategy waitStrategy,
+      List<SpanProcessor> spanProcessorList,
+      boolean blocking) {
+    // Create new Disruptor for processing. Note that Disruptor creates a single thread per
+    // consumer (see https://github.com/LMAX-Exchange/disruptor/issues/121 for details);
+    // this ensures that the event handler can take unsynchronized actions whenever possible.
+    Disruptor<DisruptorEvent> disruptor =
+        new Disruptor<>(
+            DisruptorEventFactory.INSTANCE,
+            bufferSize,
+            new ThreadFactoryWithName(WORKER_THREAD_NAME),
+            ProducerType.MULTI,
+            waitStrategy);
+    disruptor.handleEventsWith(
+        new DisruptorEventHandler[] {new DisruptorEventHandler(spanProcessorList)});
+    disruptor.start();
+
+    return new DisruptorEventQueue(disruptor, spanProcessorList, blocking);
+  }
+
+  // Enqueues an event on the {@link DisruptorEventQueue}.
+  void enqueue(ReadableSpan readableSpan, boolean isStart) {
+    if (isShutdown) {
+      if (!loggedShutdownMessage.getAndSet(true)) {
+        logger.log(Level.INFO, "Attempted to enqueue entry after Disruptor shutdown.");
+      }
+    }
+
+    if (blocking) {
+      long sequence = ringBuffer.next();
+      try {
```

Comment:
nit: this and https://github.com/open-telemetry/opentelemetry-java/pull/468/files#diff-6c02fc3d87a7b36b5f8badb3359dd315R107 can be combined.